airbyte-integrations/base-python/airbyte_protocol/integration.py

```diff
@@ -0,0 +1,76 @@
+import json
+from dataclasses import dataclass
+from typing import Generator
+import os
+
+from .models import AirbyteCatalog, AirbyteMessage
+
+
+class AirbyteSpec(object):
+    @staticmethod
+    def from_file(file):
+        with open(file) as file:
+            spec_text = file.read()
+        return AirbyteSpec(spec_text)
+
+    def __init__(self, spec_string):
+        self.spec_string = spec_string
+
+
+class AirbyteCheckResponse(object):
+    def __init__(self, successful, field_to_error):
+        self.successful = successful
+        self.field_to_error = field_to_error
+
+
+@dataclass
+class ConfigContainer:
+    raw_config: object
+    rendered_config: object
+    raw_config_path: str
+    rendered_config_path: str
+
+
+class Integration(object):
+    def __init__(self):
+        pass
+
+    def spec(self, logger) -> AirbyteSpec:
+        # test if spec is present in cwd to allow code to run outside of a docker image
+        if os.path.exists('spec.json'):
```

Comment:
This encourages different behavior during the dev process vs what is happening in the final Docker image. The functionality of python singer-based integrations should be completely self-contained at the package level. The `spec.json` file should only be loaded from the "resources" of the package.

Reply:
Yes, not happy that we have a different behavior, but because I am trying to make a convention that spec is always at the same place that should work.

I don't understand the point abut singer-based. Let's talk about offline